src/components/Composer/index.tsx

```diff
@@ -216,16 +218,88 @@ function Composer(
+
+                    // convert base64 to file blob
+                    const base64ImageContent = src.split(',')[1];
+                    const mimeType = src.split(';')[0].split(':')[1];
+                    const byteString = atob(base64ImageContent);
+                    const arrayBuffer = new ArrayBuffer(byteString.length);
+                    const uint8Array = new Uint8Array(arrayBuffer);
+                    for (let i = 0; i < byteString.length; i++) {
+                        uint8Array[i] = byteString.charCodeAt(i);
+                    }
+                    const blob = new Blob([arrayBuffer], {type: mimeType});
+                    const file = new File([blob], 'image.jpg', {type: 'image/jpeg'});
+
+                    onPasteFile(file);
+                    return;
+                }
+            }
+
+            // If the clipboardDataHtml contains google user content image link
+            if (clipboardDataHtml?.includes(CONST.GOOGLE_DOC_IMAGE_LINK_MATCH)) {
+                const domparser = new DOMParser();
+                const pastedHTML = clipboardDataHtml;
+                const embeddedImages = domparser.parseFromString(pastedHTML, TEXT_HTML).images;
+
+                if (embeddedImages.length > 0 && embeddedImages[0]?.src) {
+                    const src = embeddedImages[0].src;
+                    if (src.includes(CONST.GOOGLE_DOC_IMAGE_LINK_MATCH)) {
+                        fetch(src)
+                            .then((response) => response.blob())
+                            .then((blob) => {
+                                const file = new File([blob], 'image.jpg', {type: 'image/jpeg'});
+                                onPasteFile(file);
+                            });
+                        return;
+                    }
+                }
+            }
+
+            // If the clipboardDataHtml contains google sheets html data (pasted chart)
+            if (clipboardDataHtml?.includes(CONST.GOOGLE_SHEETS_HTML_MATCH)) {
+                const domparser = new DOMParser();
+                const pastedHTML = clipboardDataHtml;
+                const dom = domparser.parseFromString(pastedHTML, TEXT_HTML);
+                const sheetHTML = dom.getElementsByTagName('google-sheets-html-origin')?.item(0);
+                const table = sheetHTML?.getElementsByTagName('table')?.item(0);
+
+                if (sheetHTML) {
+                    document.body.appendChild(sheetHTML);
+
+                    // in case there's no table, we'll screenshot the whole pasted sheet
+                    html2canvas(table ?? (sheetHTML as HTMLElement)).then((canvas) => {
+                        const file = FileUtils.dataURItoBlob(canvas.toDataURL('image/png'));
+                        onPasteFile(file);
+                    });
+
+                    // cleanup
+                    document.body.removeChild(sheetHTML);
+                    return;
+                }
+            }
```

Comment:
This is not something we want to do here, i.e. we do not want to make any screenshots of objects.

Reply:
I see, then I will remove that part.
I suppose we're not looking at having functionality that will allow to copy html objects like tables.

Reply:
Only copy actual objects like charts.

Reply:
Understood, that's already being covered by the base64 block.
I'll clean it up right now!

Reply:
> I suppose we're not looking at having functionality that will allow to copy html objects like tables.

We would have in the future, I guess. The table should be copied as a table in mark down, that's when it makes sense